SearchService2/appscale/search/scripts.py

```diff
@@ -0,0 +1,184 @@
+"""
+This module provides helper scripts for managing Solr collections.
+"""
+import argparse
+import logging
+
+import sys
+
+from appscale.search.constants import SolrServerError, SolrClientError
+from kazoo.client import KazooClient
+from appscale.common.constants import LOG_FORMAT, ZK_PERSISTENT_RECONNECTS
+from tornado import ioloop, gen
+
+from appscale.search import solr_adapter
+
+logger = logging.getLogger(__name__)
+
+
+def list_solr_collections():
+  """ Lists all Solr collections.
+  """
+  logging.basicConfig(format=LOG_FORMAT, level=logging.INFO)
+
+  parser = argparse.ArgumentParser()
+  parser.add_argument(
+    '-v', '--verbose', action='store_true',
+    help='Output debug-level logging')
+  parser.add_argument(
+    '--zk-locations', required=True, nargs='+', help='ZooKeeper location(s)')
+  args = parser.parse_args()
+
+  if args.verbose:
+    logging.getLogger('appscale').setLevel(logging.DEBUG)
+
+  zk_client = KazooClient(
+    hosts=','.join(args.zk_locations),
+    connection_retry=ZK_PERSISTENT_RECONNECTS
+  )
+  zk_client.start()
+
+  async def list_collections():
+    """ Asynchronously fetches a list of Solr collections
+    from SolrCloud and prints it to stdout.
+    """
+    adapter = solr_adapter.SolrAdapter(zk_client)
+    try:
+      collections, broken = await adapter.solr.list_collections()
+      logger.info('Collections:\n    {}'.format('\n    '.join(collections)))
+      if broken:
+        logger.warning('Broken collections:\n    {}'.format('\n  '.join(broken)))
+    except (SolrServerError, SolrClientError) as err:
+      logger.error('Failed to list Solr collections ({}).'.format(err))
+    io_loop.stop()
+    io_loop.close()
+
+  io_loop = ioloop.IOLoop.current()
+  io_loop.spawn_callback(list_collections)
```

Comment:
Would `IOLoop.run_sync` be appropriate for this?

Reply:
Oh.. good old copy-paste=) Thanks, I'll update it